contrib_bots/lib/converter.py

```diff
@@ -0,0 +1,145 @@
+    The message '@convert help' posts a short description of how to use the plugin, along with a list of all supported units.
+    '''
+
+    units = {'bit': [0, 1, 'bit'], 'byte': [0, 8, 'bit'],
+             'kelvin': [0, 1, 'kelvin'], 'fahrenheit': [255.372222, 0.555555, 'kelvin'], 'celsius': [273.15, 1, 'kelvin'],
+             'meter': [0, 1, 'meter'], 'inch': [0, 0.0254, 'meter'], 'foot': [0, 0.3048, 'meter'],  'yard': [0, 0.9144, 'meter'],
+             'mile': [0, 1609.344, 'meter'], 'nautical-mile': [0, 1852, 'meter'],
+             'square-meter': [0, 1, 'square-meter'], 'square-kilometer': [0, 1000000, 'square-meter'],
+             'square-centimeter': [0, 0.0001, 'square-meter'], 'are': [0, 100, 'square-meter'],
+             'hectare': [0, 10000, 'square-meter'], 'square-inch': [0, 0.00064516, 'square-meter'],
+             'square-foot': [0, 0.09290304, 'square-meter'], 'square-yard': [0, 0.83612736, 'square-meter'],
+             'acre': [0, 4046.8564224, 'square-meter'], 'square-mile': [0, 2589988.110336, 'square-meter'],
+             'cubic-meter': [0, 1, 'cubic-meter'], 'cubic-centimeter': [0, 0.000001, 'cubic-meter'],
+             'cubic-decimeter': [0, 0.001, 'cubic-meter'], 'cubic-inch': [0, 0.000016387064, 'cubic-meter'],
+             'cubic-foot': [0, 0.028316846592, 'cubic-meter'], 'liter': [0, 0.001, 'cubic-meter'],
+             'cubic-yard': [0, 0.764554857984, 'cubic-meter'], 'fluid-ounce': [0, 0.000029574, 'cubic-meter'],
+             'gram': [0, 1, 'gram'], 'ton': [0, 1000000, 'gram'], 'ounce': [0, 28.349523125, 'gram'], 'pound': [0, 453.59237, 'gram']}
+
+    prefixes = {'atto': -18, 'femto': -15, 'pico': -12, 'nano': -9, 'micro': -6, 'milli': -3, 'centi': -2, 'deci': -1,
+                'deca': 1, 'hecto': 2, 'kilo': 3, 'mega': 6, 'giga': 9, 'tera': 12, 'peta': 15, 'exa': 18}
+
+    def usage(self):
+        return '''
+            This plugin allows users to make conversions between various units, e.g. Celsius to Fahrenheit, or Kilobytes to Gigabytes.
+            It looks for messages of the format '@convert <number> <original-unit> <new-unit>.
+            The message '@convert help' posts a short description of how to use the plugin, along with a list of all supported units.
+            '''
+
+    def triage_message(self, message):
+        return ('@convert' in message['content'])
+
+    def isfloat(self, value):
+        try:
+            float(value)
+            return True
+        except:
+            return False
+
+    def round_to(self, x, digits):
+        return round(x, digits-int(floor(log10(abs(x)))))
+
+    def handle_message(self, message, client, state_handler):
+        content = message['content']
+
+        content = content.lower()
+        words = content.split()
+        conversion_occ = [idx for idx, x in enumerate(copy.copy(words)) if x == "@convert"]
+        results = []
+
+        for idx in conversion_occ:
+            if (idx+1) < len(words) and words[idx+1] == 'help':
```

Comment:
We could handle it in some extra code, but I feel that this way is more elegant and keeps it simple :)